File: 800/1579A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        while(t-- > 0)
        {
            String s = sc.next();
            
            int cnt = 0;
            int cnt_b = 0;
            
            for(int i=0;i<s.length();i++)
            {
                if(s.charAt(i) == 'A' || s.charAt(i) == 'C')
                    cnt++;
                else
                    cnt_b++;
            }
            
            if(cnt_b == cnt)
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}

File: 800/1579A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        string s;
        cin >> s;

        int cnt = 0;     // count of 'A' and 'C'
        int cnt_b = 0;   // count of 'B'

        for (char c : s) 
        {
            if (c == 'A' || c == 'C')
                cnt++;
            else
                cnt_b++;
        }

        if (cnt_b == cnt)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}
